src/mruset.h

```diff
@@ -45,25 +50,19 @@ class mruset
     {
         std::pair<iterator, bool> ret = set.insert(x);
         if (ret.second) {
-            if (nMaxSize && queue.size() == nMaxSize) {
-                set.erase(queue.front());
-                queue.pop_front();
+            if (set.size() == nMaxSize + 1) {
+                set.erase(order[first_used]);
+                order[first_used] = set.end();
+                first_used++;
+                if (first_used == nMaxSize) { first_used = 0; }
```

Comment:
Modulo operator is slow, if it's not known to the compiler to be a power of two.